realm/realm-library/src/androidTest/java/io/realm/MutableRealmIntegerTests.java

```diff
@@ -152,6 +152,54 @@ public void validAndManaged_managed() {
         realm.commitTransaction();
     }
 
+    /**
+     * @Required MutableRealmIntegers should not be nullable.
+     * There are other tests testing nullabilty: just need to test @Required here.
+     * There is no attempt to control the nullability of an unmanaged MutableRealmInteger.
+     * And attempt to copy an unmanaged model object with a null-valued MutableRealmInteger
+     * into an @Required field should fail.
+     */
+    @Test
+    public void required() {
+        realm.beginTransaction();
+        MutableRealmIntegerTypes c1 = realm.createObject(MutableRealmIntegerTypes.class);
+        realm.commitTransaction();
+
+        Table table = realm.getTable(MutableRealmIntegerTypes.class);
```

Comment:
You can use `realm.getSchema().get("MutableRealmIntegerTypes").isNullable(field)` which is our public API. I would avoid using private API's in unit tests if at all possible.